kernel/CMakeLists.txt

```diff
@@ -52,7 +52,10 @@ target_sources_ifdef(
   userspace.c
   )
 
+get_property(E_KERNEL_ENTRY GLOBAL PROPERTY E_KERNEL_ENTRY)
 
 add_dependencies(kernel offsets_h)
 
-target_link_libraries(kernel zephyr_interface)
+target_link_libraries(kernel PUBLIC zephyr_interface)
+target_link_libraries(kernel INTERFACE ${E_KERNEL_ENTRY})
```

Comment:
It is better to get rid of this flag than to move it around. This PR will need to be updated when https://github.com/zephyrproject-rtos/zephyr/pull/8534 is merged.

Reply:
Agree. If the flag can be removed and the functionality be part of the linker script, that is much better.
But until then, then flag would more correctly be a link flag propagated from kernel lib to those who links to it, i.e. all applications.